packages/@react-stately/table/src/TableCollection.ts

```diff
@@ -217,6 +217,10 @@ export class TableCollection<T> extends GridCollection<T> implements ITableColle
             }
           }
           break;
+        case 'section':
```

Comment:
Note: The way we calculate index + how I conflated the `section` as a `row` bothered me a bit so I've done some (extremely rough) work [here](https://github.com/adobe/react-spectrum/commit/d8252866fb56c09126a2600260427475d595b18b) to refactor the Table collection, layout, etc so that the TableSection generates its own row + cell and the index is calculated with respect to the node's position within its parent. It does complicate the code a little bit and I can't quite settle on if it is overkill so pausing on the work there